include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,194 @@
+#include "volume/volume_cooling_balls.hpp"
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+typedef typename HPolytope<Point>::MT MT;
+typedef typename HPolytope<Point>::VT VT; 
+
+typedef const unsigned int Uint;  // positive constant value for no of samples & dimensions
+enum volType { CB , CG , SOB }; // Volume type for polytope
+
+// To check if two n-dimensional points ensure valid limits in integration 
+bool validLimits(Point LL, Point UL){
+    if( UL.dimension() == LL.dimension() ) {
+        for ( int i = 0 ; i< LL.dimension() ; i++){
+            if( LL[i] > UL[i]){
+            std::cout << "Invalid integration limits\n";
+            return false;
+            }
+        }
+        return true;
+    }else{
+        std::cout << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Hyper-Rectangle volume calculator in n-dimensions
+NT hyperRectVolume(Point LL, Point UL){
+    NT product=1;
+    if(validLimits(LL,UL)){
+        for(int i=0; i<LL.dimension(); ++i){
+            product = product * ( UL[i] - LL[i] );
+        }
+        return product;
+    }
+    else return 0;
+}
+
+// To sample a point between two n-dimensional points using inbuilt random sampling
+// Point samplerBWLimits(Point LL, Point UL){
+//     Point sample_point(LL.dimension());
+//     for(int i=0; i<LL.dimension(); ++i){
+//         sample_point.set_coord(i , LL[i] + (NT)(rand()) / ((NT)(RAND_MAX/(UL[i] - LL[i]))) );
+//     }
+//     return sample_point;
+// }
+
+// Simple MC Integration over Hyper-Rectangles
+template
+<
+    typename WalkType=BallWalk,
+    typename Functor
+>
+void simple_mc_integrate(Functor Fx, Uint N ,Point LL, Point UL, int walk_length=10, NT e=0.1){
+    Uint dim=LL.dimension();
+    NT sum = 0;
+    if(validLimits(LL,UL)){
+
+        // Creating an MT & VT for HPolytope(Hyper-Rectangle) for integration limits using LL & UL
+        MT mt(dim*2,dim);
+        VT vt(dim*2);
+        for(int i=0 ; i<dim ; i++){
+            mt(i,i)=1;
+            vt(i)=UL[i];
+            mt(dim+i,i)=-1;
+            vt(dim+i)=LL[i]*-1;
+        }
+
+        // Initialization of H-Polytope and setting up params for random walks
+        HPOLYTOPE P(dim,mt,vt);
+        // P.print();
+        std::pair<Point, NT> inner_ball = P.ComputeInnerBall();
+        RandomNumberGenerator rng(1);
+        Point x0 = inner_ball.first;
+        typename WalkType::template Walk<HPOLYTOPE, RandomNumberGenerator> walk(P,x0,rng);
+        
+        for (int i = 0; i <=N; i++ ) {
+            walk.apply(P,x0,walk_length,rng);
+            sum = sum + Fx(x0);
+        }    
+        NT volume = hyperRectVolume(LL,UL);
+        std::cout << "Volume of the subspace: " << volume << std::endl;
```

Comment:
Avoid having print statements within code that is in a header file, that is remove them whenever it is not necessary.